### ISSUE TITLE
[Bug] Transaction with custom Chialisp remains stuck pending

### ISSUE THREAD

**Issue by ccornick** [Chia-Network/chia-blockchain#13970]
### What happened?

Custom Chialisp was written to enable tokenization of carbon credits on Chia. When a specific custom TAIL is used by a climate CAT, the transaction gets stuck in 'pending' status, despite the transaction going through correctly.

### Version

Latest Dev 11/18/2022

### What platform are you using?

Windows

### What ui mode are you using?

GUI

**Comment by Quexington** [comment]
What is the nature of this transaction? Are you running the TAIL somehow?  Is this a mint/melt? Or is it just a standard transaction from one party to another of a CAT with this TAIL?

**Comment by ccornick** [comment]
The nature of the transaction is that we have one entity (A) that creates a CAT and applies custom TAIL. That CAT is sent to a separate entity (B).

Entity B may prefer to revert the CAT back into the asset it represented. This requires entity B to request entity A melts the CAT directly from entity B's wallet.

So the transaction is getting sent from entity A, to melt something directly from entity B, and entity A's wallet gets the stuck transaction. Entity B is unaffected.

**Comment by emlowe** [comment]
During bug scrub, Sebastian mentioned that our current wallet doesn't know about this TAIL and so there may be some more code needed to be modified for this special wallet. Although this sounds like in general the CAT works in that is appears in the wallet and the problem is with this specific melt transaction. (So this TAIL allows the minter/creator to melt even if they don't own the cat?)

**Comment by ccornick** [comment]
Makes sense that the current wallet doesn't know about this TAIL, it was custom built by our friends at Hashgreen within the past month or so.

You are spot on with your understanding. Minter/creator can melt even if they do not own the CAT. It requires the CAT owner to initiate a request and then the minter completes that request to melt the CAT directly from the holder wallet.

**Comment by Quexington** [comment]
Since this is something way out of the scope of our existing wallet, I'm going to assume there's custom code generating this tx record that's being pushed.  Make sure you've filled out all of the fields on the tx record correctly especially additions and removals.

**Comment by Quexington** [comment]
@ccornick do you have any update on this issue?  I think I heard that it was not 100% of the time reproducible.

**Comment by Quexington** [comment]
@ccornick is this still an unresolved issue?

**Comment by ccornick** [comment]
No, I can't reproduce the issue anymore. Good to close.
